drivers/npmx/npmx_shell.c

```diff
@@ -1792,6 +1798,176 @@ static int cmd_adc_meas_take_vbat(const struct shell *shell, size_t argc, char *
+		switch (config_type) {
+		case POF_CONFIG_TYPE_ENABLE:
+			shell_print(shell, "Value: %d.", (int)pof_config.status);
+			break;
+		case POF_CONFIG_TYPE_POLARITY:
+			shell_print(shell, "Value: %d.", (int)pof_config.polarity);
+			break;
+		case POF_CONFIG_TYPE_THRESHOLD:
+			uint32_t mVolts;
+			if (npmx_pof_threshold_convert_to_mv(pof_config.threshold, &mVolts)) {
+				shell_print(shell, "Value: %d mV.", mVolts);
+			} else {
+				shell_error(shell,
+					    "Error: unable to convert threshold value to mV.");
+			}
+			break;
+		}
+	} else {
+		shell_error(shell, "Error: unable to read POF config.");
+	}
+
+	return 0;
+}
+
+static int pof_config_set(const struct shell *shell, size_t argc, char **argv,
+			  pof_config_type_t config_type)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	if (argc < 2) {
+		switch (config_type) {
+		case POF_CONFIG_TYPE_ENABLE:
+			shell_error(shell, "Error: missing enable value.");
+			return 0;
+		case POF_CONFIG_TYPE_POLARITY:
+			shell_error(shell, "Error: missing polarity value.");
+			return 0;
+		case POF_CONFIG_TYPE_THRESHOLD:
+			shell_error(shell, "Error: missing threshold value.");
+			return 0;
+		default:
+			shell_error(shell, "Error: invalid config type.");
+			return 0;
+		}
+	}
+
+	int err = 0;
+	uint32_t value = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT32_MAX);
+
+	if (err != 0) {
+		shell_error(shell, "Error: value must be an integer.");
+		return 0;
+	}
+
+	switch (config_type) {
+	case POF_CONFIG_TYPE_ENABLE:
+		if (value >= NPMX_POF_STATUS_COUNT) {
+			shell_error(shell, "Error: enable value can be 0-Disable, 1-Enable.");
+			return 0;
+		}
+		break;
+	case POF_CONFIG_TYPE_POLARITY:
+		if (value >= NPMX_POF_POLARITY_COUNT) {
+			shell_error(shell,
+				    "Error: polarity value can be 0-Active low, 1-Active high.");
+			return 0;
+		}
+		break;
+	case POF_CONFIG_TYPE_THRESHOLD:
+		if (npmx_pof_threshold_convert(value) == NPMX_POF_THRESHOLD_INVALID) {
+			shell_error(
+				shell,
+				"Error: threshold value can be between 2600 and 3500, in increments of 100.");
+			return 0;
+		}
+		break;
+	default:
+		shell_error(shell, "Error: invalid config type.");
+		return 0;
+	}
+
+	npmx_pof_t *pof_instance = npmx_pof_get(npmx_instance, 0);
+
+	npmx_pof_config_t pof_config;
+	npmx_error_t err_code = npmx_pof_config_get(pof_instance, &pof_config);
+
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to get POF config.");
+	}
+
+	switch (config_type) {
+	case POF_CONFIG_TYPE_ENABLE:
+		pof_config.status = (value ? NPMX_POF_STATUS_ENABLE : NPMX_POF_STATUS_DISABLE);
+		break;
+	case POF_CONFIG_TYPE_POLARITY:
+		pof_config.polarity = (value ? NPMX_POF_POLARITY_HIGH : NPMX_POF_POLARITY_LOW);
+		break;
+	case POF_CONFIG_TYPE_THRESHOLD:
+		pof_config.threshold = npmx_pof_threshold_convert(value);
```

Comment:
Should be checked if returned value is not invalid. `npmx_pof_config_set` would fall in assert if it was, but we try to avoid using assertion in shell and print error message instead.

Reply:
Already done here: https://github.com/NordicSemiconductor/npmx-zephyr/pull/56/files#diff-0e688e7fadaa6f2edadba6ab36e3e5be7963e487870b190f00eeceda2c0fc626R1890

Reply:
Then maybe result of conversion could be saved to a variable in the switch-case? I am not a fan of calling the same function twice.